Source/IntegrationTests/TestFiles/LitTests/LitTest/invariants/Counter.dfy

```diff
@@ -0,0 +1,20 @@
+// RUN: %verify --type-system-refresh --check-invariants "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+// Store the old(value) since we don't yet have twostate invariants
+class Counter {
+  ghost var old_value: nat
+  ghost var value: nat
+  constructor() {
+    old_value := 0;
+    value     := 0;
+  }
+  method Increment()
+    modifies this
+  {
+    old_value := value;
+    value     := value + 1;
+  }
+  invariant value == old_value + 1
```

Comment:
How come there is no complaint about this invariant, since the condition does not hold at the end of the constructor?